apps/application-system/api/src/app/modules/payment/payment.service.ts

```diff
@@ -66,22 +66,24 @@ export class PaymentService {
       this.paymentConfig.callbackAdditionUrl +
       payment.id
 
-    const parsedDefinition = JSON.parse(
-      (payment.definition as unknown) as string,
-    )
+    const parsedDefinition =
+      typeof payment.definition === 'string' // this is a hack to get around the fact that the definition is stored as a string in the database but as an object in during testing
+        ? JSON.parse((payment.definition as unknown) as string)
+        : JSON.parse(JSON.stringify(payment.definition))
```

Comment:
I would've thought that `JSON.parse(JSON.stringify(someObj)` is equivalent to `someObj`?
Am I missing something?

Reply:
I may be thinking this backwards if the norm is that the definition is a string whoops